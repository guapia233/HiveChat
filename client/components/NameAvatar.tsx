import { Avatar } from "@mui/material";

// A Google-style avatar with an initial and a color

// Copied from MUI Avatar docs
function stringToColor(string: string) {
  let hash = 0;

  for (let i = 0; i < string.length; i += 1) {
    hash = string.charCodeAt(i) + ((hash << 5) - hash);
  }

  let color = "#";

  for (let i = 0; i < 3; i += 1) {
    const value = (hash >> (i * 8)) & 0xff;
    color += `00${value.toString(16)}`.slice(-2);
  }

  return color;
}

export default function NameAvatar({ name }: { name: string }) {
  return (
    <Avatar
      sx={{
        bgcolor: stringToColor(name),
      }}
    >
      {name[0]}
    </Avatar>
  );
}
