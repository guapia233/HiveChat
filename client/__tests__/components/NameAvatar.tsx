import { render, screen } from "@testing-library/react";
import NameAvatar from "@/components/NameAvatar";
import "@testing-library/jest-dom";

describe("NameAvatar", () => {
  test.each([
    ["John", "J"],
    ["Doe", "D"],
    ["Fran", "F"],
  ])("Renders the initial and color for %s", (name, initial) => {
    // Render
    const { asFragment } = render(<NameAvatar name={name} />);

    // Sanity check
    expect(screen.getByText(initial)).toBeInTheDocument();

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });
});
