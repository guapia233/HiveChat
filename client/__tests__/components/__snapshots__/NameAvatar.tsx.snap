// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameAvatar Renders the initial and color for Doe 1`] = `
<DocumentFragment>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-jindr0-MuiAvatar-root"
  >
    D
  </div>
</DocumentFragment>
`;

exports[`NameAvatar Renders the initial and color for Fran 1`] = `
<DocumentFragment>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-yjazcl-MuiAvatar-root"
  >
    F
  </div>
</DocumentFragment>
`;

exports[`NameAvatar Renders the initial and color for John 1`] = `
<DocumentFragment>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1u97j49-MuiAvatar-root"
  >
    J
  </div>
</DocumentFragment>
`;
